src/ORM/Locator/TableContainerDelegate.php

```diff
@@ -0,0 +1,60 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.3.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\ORM\Locator;
+
+use Cake\ORM\Table;
+use Psr\Container\ContainerInterface;
+
+class TableContainerDelegate implements ContainerInterface
+{
+    use LocatorAwareTrait;
+
+    /**
+     * @var list<string>
+     */
+    protected array $cache = [];
+
+    /**
+     * @inheritDoc
+     */
+    public function get(string $id): Table
+    {
+        return $this->fetchTable($id);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function has(string $id): bool
+    {
+        if (!str_ends_with($id, 'Table')) {
+            return false;
+        }
+
+        if (in_array($id, $this->cache)) {
+            return true;
+        }
```

Comment:
This might not be a real optimization depending on the app and the cost of is_subclass_of.